internal/common/physics/engine.go

```diff
@@ -51,16 +51,12 @@ func ComputeDistance(src utils.Coordinates, target utils.Coordinates) float64 {
 
 // This function is to be called from the server only
 func GenerateNewState(initialState utils.PhysicalState, force float64, orientation float64) utils.PhysicalState {
+	var finalState utils.PhysicalState = initialState
 	acceleration := CalcAcceleration(force, initialState.Mass, initialState.Velocity)
 	velocity := CalcVelocity(acceleration, initialState.Velocity)
 	coordinates := GetNewPosition(initialState.Position, velocity, orientation)
-
-	finalState := utils.PhysicalState{
-		Position:     coordinates,
-		Acceleration: acceleration,
-		Velocity:     velocity,
-		Mass:         initialState.Mass,
-	}
-
+	finalState.Position = coordinates
```

Comment:
We have merged iteration2 where the changes were already done so i would just leave it as it is in main